src/wallet/api/swapService.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import config from "../../config";
+import axios from "axios";
+import { TIMEOUT, } from '../../constants';
+import sha256 from 'crypto-js/sha256';
+import AES from 'crypto-js/aes';
+import CryptoJS from 'crypto-js';
+
+/**
+ * Encrypts a string ( PartialTx or Signatures ) before sending it to the backend
+ * @param serialized
+ * @param password
+ */
+export function encryptString(serialized: string, password: string): string {
+  const aesEncryptedObject = AES.encrypt(serialized, password);
+
+  // Serializing the cipher output in Base64 to avoid loss of encryption parameter data
+  const baseEncryptedObj = CryptoJS.enc.Base64.parse(aesEncryptedObject.toString());
+  return CryptoJS.enc.Base64.stringify(baseEncryptedObj);
+}
+
+/**
+ * Decrypts a string ( PartialTx or Signatures ) from the backend
+ * @param serialized
+ * @param password
+ */
+export function decryptString(serialized: string, password: string): string {
+  const aesDecryptedObject = AES.decrypt(serialized, password);
+  return aesDecryptedObject.toString(CryptoJS.enc.Utf8);
```

Comment:
Why not use base64 on both encrypt/decrypt?

Reply:
We only need `Base64` encoding while the contents are encrypted. As soon as we decrypt the contents, we need it back in `Utf8` so that we can work with it in javascript code. So the final decryption on line 35 should always be to `Utf8`.

If we didn't do this string encoding here, all places that need to use the decrypted object would have to implement the Base64 -> Utf8 conversion themselves.

Reply:
If your question is related to the line 34, it automatically detects the serialized contents' encoding.